src/EditorFeatures/CSharpTest/Diagnostics/MakeMethodAsynchronous/MakeMethodAsynchronousTests.cs

```diff
@@ -599,5 +599,180 @@ class Program
 }";
             await TestInRegularAndScriptAsync(initial, expected);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodAsynchronous)]
+        [WorkItem(14133, "https://github.com/dotnet/roslyn/issues/14133")]
+        public async Task AddAsyncInLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Threading.Tasks;
+
+class C
+{
+    public void M1()
+    {
+        void M2()
+        {
+            [|await M3Async();|]
+        }
+    }
+
+    async Task<int> M3Async()
+    {
+        return 1;
+    }
+}",
+@"using System.Threading.Tasks;
+
+class C
+{
+    public void M1()
+    {
+        async Task M2Async()
+        {
+            await M3Async();
+        }
+    }
+
+    async Task<int> M3Async()
+    {
+        return 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodAsynchronous)]
+        [WorkItem(14133, "https://github.com/dotnet/roslyn/issues/14133")]
+        public async Task AddAsyncInLocalFunctionKeepVoidReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Threading.Tasks;
+
+class C
+{
+    public void M1()
+    {
+        void M2()
+        {
+            [|await M3Async();|]
+        }
+    }
+
+    async Task<int> M3Async()
+    {
+        return 1;
+    }
+}",
+@"using System.Threading.Tasks;
+
+class C
+{
+    public void M1()
+    {
+        async void M2Async()
+        {
+            await M3Async();
+        }
+    }
+
+    async Task<int> M3Async()
+    {
+        return 1;
+    }
+}",
+index: 1);
+        }
+
+        [Theory]
```

Comment:
i'm really not liking this Theory approach to these sorts of tests.  Reading the tests is now much harder.  And it makes basic stuff like { } in the test become distracting as {{ and }}   I don't see why this is actually better than just having two actual tests here.  One for the Task case and one for the void case.

A real purpose for these tests is for the text of the test to be self contained and self describing.  i.e. a single string that contains all the information necessary to know what's going on.  That's why we have the markup language.  There is no separate mechanism for specifying the caret position.  It's just in the text.  This 'Theory' approach fights against that, with negative results IMO.

I think theories can be good for very basic, non-replacing tests.  Where the theory/data contain all the information you need to know.  In that sense, it's a way of stamping out tests easily, but each test is still clear on what it does.  As that's not the case here 👎 on this part for me.